document `--from-commit` and `--include-easyblocks-from-commit` (#274)
easybuilders/easybuild-docs: docs/integration-with-github.md

Again, note that locally available easyconfigs that are required to
resolve dependencies are being picked up as needed.

## Using easyconfigs from a commit {: #github_from_commit }

*(`--from-commit`, supported since EasyBuild v4.9.1)*

Using `--from-commit`, you can let EasyBuild use easyconfig files that were added or modified in a specific Git commit.

This is similar to using `--from-pr` (see [Using easyconfigs from pull requests][github_from_pr]), but it differs in two ways:

- The GitHub API is *not* used to obtain the easyconfig files from a particular commit. This can be important since
  it avoids hitting the [GitHub API rate limits](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api),
  especially when no GitHub token is available;

> - The easyconfig files will be *exactly* those as they are in the specified commit,

Reviewer: ```suggestion
- The easyconfig files will be *exactly* as they are in the specified commit
```

>   as they can also be consulted [on GitHub](https://github.com/easybuilders/easybuild-easyconfigs/tree/c0ff3315c0ffeec0ff3315c0ffeec0ffeec0ff33/easybuild/easyconfigs)

Reviewer: ```suggestion
  and they can also be consulted [on GitHub](https://github.com/easybuilders/easybuild-easyconfigs/tree/c0ff3315c0ffeec0ff3315c0ffeec0ffeec0ff33/easybuild/easyconfigs)
```

>   (link points to a fictious commit SHA `c0ff3315c0ffeec0ff3315c0ffeec0ffeec0ff33`).

Reviewer: ```suggestion
  (link points to a fictitious commit SHA `c0ff3315c0ffeec0ff3315c0ffeec0ffeec0ff33`).
```

  This is not the case with `--from-pr`, since then either the PR branch is merged with the target branch
  (typically `develop`) before obtained the easyconfigs (for open PRs), or the  `develop` branch is used (for merged PRs).
  This aspect can be important to ensure reproducibility.

!!! note
    The commit SHA passed to `--from-commit` must be a full commit SHA consisting of 40 characters,

>     not a 10-character shortered commit SHA as you often see in GitHub.

Reviewer: ```suggestion
    not a the first part of a commit SHA as you often see in GitHub.
```

Example usage:

> ```shell

Reviewer: ```suggestion

```shell
```
